### DISCUSSION TITLE
Spatial Containment Widget

### DISCUSSION THREAD

**Discussion by xihumeijing** [iTwin/itwinjs-core#5837] [Q&A] [answered]
![a](https://github.com/iTwin/itwinjs-core/assets/19543763/92620e12-ac5f-4f23-a885-08773d6bc063)

When I was browsing the model online on the iTwin Platform, I found a widget that views the model by floor. Please refer to the image. Where is the component code for this widget？

**Comment by grigasp** [comment, marked as answer]
Hi. The package is called `@itwin/breakdown-trees-react` and its source is located [here](https://github.com/iTwin/viewer-components-react/blob/master/packages/itwin/breakdown-trees/package.json).

**Reply by xihumeijing** [reply]
![image](https://github.com/iTwin/itwinjs-core/assets/19543763/fbbc3567-b851-4699-b469-e059f58a28e9)
![image](https://github.com/iTwin/itwinjs-core/assets/19543763/9cf2c1bf-0ffe-4305-8647-dfab735b81c6)

My project version is 3. Can I only upgrade to 4.0 to apply the current @itwin/breakdown-trees-react

**Reply by grigasp** [reply]
You got two options:
1. Upgrade your itwinjs-core dependencies to 4.x (preferred).
2. Use an older version of breakdown-trees-react that's compatible with core 3.5 that you're using. As far as I can tell the last version compatible with 3.x is [`0.3.16`](https://www.npmjs.com/package/@itwin/breakdown-trees-react/v/0.3.16?activeTab=code).
